editor/plugins/animation_library_editor.cpp

```diff
@@ -0,0 +1,689 @@
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "animation_library_editor.h"
+#include "editor/editor_file_dialog.h"
+#include "editor/editor_node.h"
+#include "editor/editor_scale.h"
+
+void AnimationLibraryEditor::set_animation_player(Object *p_player) {
+	player = p_player;
+}
+
+void AnimationLibraryEditor::_add_library() {
+	add_library_dialog->set_title(TTR("Library Name:"));
+	add_library_name->set_text("");
+	add_library_dialog->popup_centered();
+	add_library_name->grab_focus();
+	adding_animation = false;
+	adding_animation_to_library = StringName();
+	_add_library_validate("");
+}
+
+void AnimationLibraryEditor::_add_library_validate(String name) {
+	String error;
+
+	if (adding_animation) {
+		Ref<AnimationLibrary> al = player->call("get_animation_library", adding_animation_to_library);
+		ERR_FAIL_COND(al.is_null());
+		if (name == "") {
+			error = TTR("Animation name can't be empty.");
+
+		} else if (String(name).contains("/") || String(name).contains(":") || String(name).contains(",") || String(name).contains("[")) {
+			error = TTR("Animation name contains invalid characters.");
```

Comment:
```suggestion
			error = TTR("Animation name contains invalid characters: '/', ':', ',' or '['.");
```
Also, should `]` be forbidden too for good measure?

Reply:
Nevermind it was added to avoid user overriding internal animations like [stop], so imo that should be good enough.